### PR TITLE
Add data uri support for legend URLs

### PR DESCRIPTION
**Pull request by hwbllmnn**
External graphics in styles already support data uris since #555 and #557 

Did I already sign the contributor agreement last time?

### CODE REVIEW

**Comment by sbrunner** [comment]
The build on master is actually broken, don't worry about it...

**Comment by sbrunner** [comment]
Can you rebase this pull request, the master is now fixed :-)

**Comment by hwbllmnn** [comment]
> Can you rebase this pull request, the master is now fixed :-)

Done. But CI is still failing, I guess because of some missing credentials in my fork's settings?

**Comment by sbrunner** [comment]
No, it just never tested with a fork, I will test it on my own :-)

**Comment by sbrunner** [comment]
Can you again do a rebase, now the tests should work from a fork :-)

**Comment by hwbllmnn** [comment]
Looks like the legend example test fails now due to your changes to the request and/or the test setup. Unfortunately I don't have access to the image you're using. When trying to `docker-compose up -d` I get:

```
Pulling print (mapfish_print_tester:)...
ERROR: pull access denied for mapfish_print_tester, repository does not exist or may require 'docker login': denied: requested access to the resource is denied
```

Can you update the expected legend image or give me access to the container? At least the credentials issues seem to be fixed ;-)

**Comment by sbrunner** [comment]
Before doing `docker-compose up -d` you should do `make build` and the missing images will be built :-)

File: examples/src/test/resources/examples/legend/requestData.json [hunk 1]
```diff
@@ -28,7 +28,10 @@
                 "icons": ["http://geoserver:8080/www/legends/legend3.png"]
             }, {
                 "name": "Alea",
-                "icons": ["http://geoserver:8080/www/legends/legend4.png"]
+                "icons": ["http://localhost:8090/www/legends/legend4.png"]
```
Comment (sbrunner) [inline examples/src/test/resources/examples/legend/requestData.json:31]:
```suggestion
                "icons": ["http://geoserver:8080/www/legends/legend4.png"]
```

Comment (sbrunner) [reply examples/src/test/resources/examples/legend/requestData.json:31]: With that the CI should work :-)

**Comment by hwbllmnn** [comment]
> Before doing `docker-compose up -d` you should do `make build` and the missing images will be built :-)

Ah ok. Unfortunately that gives me

```
Step 9/17 : COPY gradle.properties build.gradle settings.gradle CI.asc /src/
COPY failed: stat /var/lib/docker/tmp/docker-builder207450754/CI.asc: no such file or directory
```

**Comment by sbrunner** [comment]
OK, I will fix that... (just add a touch CI.asc :-) )

**Comment by sbrunner** [comment]
Thanks :-)

**Pull request merged**